lib/ui/form/widget/form_survey_answer_text_field.dart

```diff
@@ -1,64 +1,68 @@
-import 'package:collection/collection.dart';
 
 class FormSurveyAnswerTextField extends StatefulWidget {
-  final QuestionModel question;
-  const FormSurveyAnswerTextField({super.key, required this.question});
+  final List<AnswerModel> answers;
+  final ValueChanged<List<SubmitSurveyAnswerModel>> onUpdateAnswer;
+
+  const FormSurveyAnswerTextField({
+    super.key,
+    required this.answers,
+    required this.onUpdateAnswer,
+  });
 
   @override
   State<FormSurveyAnswerTextField> createState() =>
       _FormSurveyAnswerTextFieldState();
 }
 
 class _FormSurveyAnswerTextFieldState extends State<FormSurveyAnswerTextField> {
-  final List<TextEditingController> _controllers = [];
-  late List<String> _textFieldHints = [];
+  late List<SubmitSurveyAnswerModel> _answerModels = [];
 
   @override
   void initState() {
     super.initState();
-    _textFieldHints =
-        widget.question.answers.map((element) => element.text).toList()
-          ..forEach((element) {
-            _controllers.add(TextEditingController());
-          });
+    _answerModels = widget.answers
+        .map((element) => SubmitSurveyAnswerModel(
+              id: element.id,
+            ))
+        .toList();
   }
 
   @override
   Widget build(BuildContext context) {
     return SingleChildScrollView(
       child: Column(
         mainAxisAlignment: MainAxisAlignment.center,
-        children: [
-          for (final textAndControllerPair in IterableZip([
-            _textFieldHints,
-            _controllers,
-          ]))
-            Padding(
-              padding: const EdgeInsets.only(bottom: AppDimensions.spacing20),
-              child: customTextField(
-                context: context,
-                controller: textAndControllerPair[1] as TextEditingController,
-                textInputType: textAndControllerPair[0] as String == 'Email'
-                    ? TextInputType.emailAddress
-                    : TextInputType.text,
-                isObscuredText: false,
-                hintText: textAndControllerPair[0] as String,
+        children: _answerModels
+            .asMap()
+            .entries
+            .map(
+              (answerEntry) => Padding(
+                padding: const EdgeInsets.only(bottom: AppDimensions.spacing20),
+                child: customTextField(
+                  context: context,
+                  controller: null,
+                  textInputType: widget.answers[answerEntry.key].text == 'Email'
+                      ? TextInputType.emailAddress
+                      : TextInputType.text,
+                  isObscuredText: false,
+                  hintText: widget.answers[answerEntry.key].text,
+                  onChanged: (text) {
+                    _answerModels[answerEntry.key] = SubmitSurveyAnswerModel(
+                      id: _answerModels[answerEntry.key].id,
+                      answer: text,
+                    );
+                    widget.onUpdateAnswer(_answerModels);
```

Comment:
can we simplify by using `widget.answers.mapIndexed`?

```suggestion
        children: widget.answers
            .mapIndexed(
              (index, answer) => Padding(
                padding: const EdgeInsets.only(bottom: AppDimensions.spacing20),
                child: customTextField(
                  context: context,
                  controller: null,
                  textInputType: answer.text == 'Email'
                      ? TextInputType.emailAddress
                      : TextInputType.text,
                  isObscuredText: false,
                  hintText: answer.text,
                  onChanged: (text) {
                    _answerModels[index] = SubmitSurveyAnswerModel(
                      id: answer.id,
                      answer: text,
                    );
                    widget.onUpdateAnswer(_answerModels);
```